admin_ui/src/utils.ts

```diff
@@ -106,32 +127,43 @@ export function convertFormValue(params: {
         value = null
     } else if (schema?.properties[key].type == "array") {
         value = JSON.parse(String(value))
-    } else if (schema?.properties[key].format == "uuid" && value == "") {
+    } else if (schema?.properties[key].format == "uuid" &&
+        schema?.properties[key].extra["nullable"] == true
+        && value == "") {
         value = null
-    } else if (schema?.properties[key].format == "email" && value == "") {
+    } else if (schema?.properties[key].format == "email" &&
+        schema?.properties[key].extra["nullable"] == true
+        && value == "") {
         value = null
-    } else if (schema?.properties[key].format == "date-time" && value == "") {
+    } else if (schema?.properties[key].format == "date-time" &&
+        schema?.properties[key].extra["nullable"] == true
+        && value == "") {
         value = null
-    } else if (schema?.properties[key].format == "date" && value == "") {
+    } else if (schema?.properties[key].format == "date" &&
+        schema?.properties[key].extra["nullable"] == true
+        && value == "") {
         value = null
-    } else if (schema?.properties[key].type == "integer" && value == "") {
+    } else if (schema?.properties[key].format == "json" &&
+        schema?.properties[key].extra["nullable"] == true
+        && value == "") {
         value = null
-    } else if (schema?.properties[key].type == "number" && value == "") {
+    }
+    else if (schema?.properties[key]["anyOf"][0].type == "integer" &&
```

Comment:
I wonder if it's possible to dramatically simplify this logic. I wonder if there are any column types left now, which when nullable and have an empty string as a value aren't converted to `null`?

Reply:
Maybe we can use a few variables to reduce code repetition. Something like this
```javascript
export function convertFormValue(params: {
    key: string
    value: any
    schema: Schema
}): any {
    let { key, value, schema } = params

    const columnTypeAnyOf = schema?.properties[key]["anyOf"][0].type
    const columnFormat = schema?.properties[key].format
    const columnNullable = schema?.properties[key].extra["nullable"] == true
    const possibleFormats = [
        "uuid",
        "email",
        "date-time",
        "date",
        "json",
        "integer",
        "number",
        "string"
    ]

    if (value == "null") {
        value = null
    } else if (schema?.properties[key].type == "array") {
        value = JSON.parse(String(value))
    } else if (possibleFormats.includes(columnFormat) &&
        columnNullable
        && value == "") {
        value = null
    } else if (possibleFormats.includes(columnTypeAnyOf) &&
        columnNullable
        && value == "") {
        value = null
    } else if (schema?.properties[key].extra.foreign_key == true &&
        columnNullable &&
        value == ""
    ) {
        value = null
    }
    return value
}
```